app/components/task_list_items/conditions_component.rb

```diff
@@ -2,28 +2,29 @@
 
 module TaskListItems
   class ConditionsComponent < TaskListItems::BaseComponent
-    def initialize(planning_application:)
-      @planning_application = planning_application
+    def initialize(condition_set:)
+      @condition_set = condition_set
     end
 
     private
 
-    attr_reader :planning_application
+    attr_reader :condition_set
+    delegate :planning_application_id, to: :condition_set
 
     def link_text
       "Add conditions"
     end
 
     def link_path
-      planning_application_conditions_path(@planning_application)
+      planning_application_conditions_path(planning_application_id)
```

Comment:
Can't we use `planning_application` here instead of `planning_application_id`? (With the delegate changed to match, of course.)

Or doesn't delegate work for activerecord relations like that?